@kiva/kv-components/package.json

```diff
@@ -63,6 +63,8 @@
     "embla-carousel": "^4.5.3",
     "focus-trap": "^6.7.2",
     "fuse.js": "^6.6.2",
+    "lodash.groupby": "^4.6.0",
```

Comment:
Instead of using groupby, could you please refactor [this function](https://github.com/kiva/kv-ui-elements/pull/309/files#diff-8401c4f3d2b2f656dcb16c9dda8b1aa1d3ed72de71117384bd59cb72024f38bbR39-R48) into a generic util and use that instead?